cpp/iid/permutation_tests.h

```diff
@@ -565,12 +564,40 @@ bool permutation_tests(const data_t *dp, const double rawmean, const double medi
 					}
 					passed_count = 0;
 					for(unsigned int j=0; j < num_tests; j++) if(!test_status[j]) passed_count++;
+                    //printf("Passed count: %d\n", passed_count);
 				}
 			}
+
+            #pragma omp atomic
+            completed ++;
+
+            if(verbose){
+                /* Construct pretty output regardless of whether on terminal (tty) or 
+                 * redirected to another file descriptor (eg. redirect to file).
+                 * Note that if using something like 'tee' to replicate the output
+                 * then it might be handy to use 'unbuffer' to fake the call into
+                 * thinking it is still being sent to a tty.
+                 */
+                if(isatty(STDOUT_FILENO))
+                    printf("\r");
+
+                #pragma omp critical
+                printf("%.02f%%: Permutuation test", (100.0*completed/PERMS));
+
+                /* If not diplaying to screen, then we can print even more information. Ultimately
+                 * we want the '\n' however printed when not printing to terminal so that the redirected
+                 * output looks nicer. 
+                 */
+                if(!isatty(STDOUT_FILENO))  {
+                    printf(" (Core %d/%d, passed_count %d)\n", omp_get_thread_num(), omp_get_num_threads()-1, passed_count);
+                }
+                fflush(stdout);
+			}
```

Comment:
I don't understand the tabbing convention here. Are these intended inconsistently with the rest of the code, or is this a github display problem?

Reply:
I think that you should set a boolean earlier in the function rather than call isatty every time. This isn't likely to change.

It's probably a better idea to build up a string internally, and then print it in a single statement within some sort of protected section (so that we don't get interlaced lines of output).